### PR TITLE
PROTOCOLS-117 Mailbox delimiter is forbidden in user name

### PR DESCRIPTION
**Pull request by aduprat**

### CODE REVIEW

File: core/pom.xml [hunk 1]
```diff
@@ -34,10 +34,14 @@
 
     <dependencies>
         <dependency>
-            <groupId>org.apache.james</groupId>
+            <groupId>${project.groupId}</groupId>
             <artifactId>james-server-util-java8</artifactId>
         </dependency>
         <dependency>
+            <groupId>${project.groupId}</groupId>
+            <artifactId>apache-james-mailbox-api</artifactId>
```
Comment (mbaechler) [inline core/pom.xml]: wrong ordering ?

File: core/src/main/java/org/apache/james/core/User.java [hunk 2]
```diff
@@ -23,12 +23,17 @@
...
+import org.apache.james.mailbox.model.MailboxConstants;
+
 import com.google.common.base.MoreObjects;
 import com.google.common.base.Preconditions;
 import com.google.common.base.Splitter;
 import com.google.common.base.Strings;
 
 public class User {
+
+    private static final String DEFAULT_DELIMITER = String.valueOf(MailboxConstants.DEFAULT_DELIMITER);
```
Comment (mbaechler) [inline core/src/main/java/org/apache/james/core/User.java]: we should make sure Default is the only possible value in James and forbid others, don't you think ?

Comment (aduprat) [reply core/src/main/java/org/apache/james/core/User.java]: I'm on it, put it's triggering a lot of modifications.
I will propose another commit for that.

File: core/pom.xml [hunk 3]
```diff
@@ -34,7 +34,11 @@
 
     <dependencies>
         <dependency>
-            <groupId>org.apache.james</groupId>
+            <groupId>${project.groupId}</groupId>
```
Comment (chibenwa) [inline core/pom.xml]: I'm really not fan of thisdependency....

It means I can't use core models in mailbox-api. Isn't it a bit sad?

I would prefer to reverse that dependency: Have an interface holding the delimitier in James core... We could then re-use it in MailboxConstants.

Comment (aduprat) [reply core/pom.xml]: For me it's the name of the `james-core` project which is wrong.
It's more `james-base-pojos`, which is the truth (only `MailAddress` & `User` class in this project).
Then, it's logic that the API depends on this project.

Comment (chibenwa) [reply core/pom.xml]: Your POJO project depends on an API project.

What if later on I want to use the pojos in my api? Or in my API package for my basic API logic?

Comment (aduprat) [reply core/pom.xml]: The `User` class needs information from the `MailboxConstants` class, which is in _apache-james-mailbox-api_ ...
I will try to put all the common classes in the _james-core_ project.

Comment (chibenwa) [reply core/pom.xml]: Maybe that constant should not be in MailboxConstants class, no?

I mean it starts having widder impact that "Just being a MailboxConstants".

Maybe we can introduce a "JamesConstants" class?

File: server/protocols/webadmin/webadmin-data/src/test/java/org/apache/james/webadmin/routes/UsersRoutesTest.java [hunk 4]
```diff
@@ -133,6 +133,31 @@ public void postShouldReturnUserErrorWhenEmptyJsonBody() {
...
+            .when()
+                .put("user.name@" + DOMAIN)
+            .then()
+                .statusCode(400);
+
+            // No user created
+            List<Map<String, String>> users =
+                when()
+                    .get()
+                    .then()
```
Comment (chibenwa) [inline server/protocols/webadmin/webadmin-data/src/test/java/org/apache/james/webadmin/routes/UsersRoutesTest.java]: Ident here ;-)

**Comment by aduprat** [comment]
#1116

**Pull request closed without merging**